From 63b0ab546832915fd0b87d94933c32e96c215649 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Fri, 21 Mar 2025 17:07:01 +0100
Subject: [PATCH 12/12] clutter/frame-clock: Don't accidentally reschedule-now

The state should go from 'dispatched-one-and-scheduled-later' to
'scheduled-later', not 'scheduled-now' when being notified about a frame
being ready - otherwise we'll dispatch without proper pacing.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4334>
---
 clutter/clutter/clutter-frame-clock.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clutter/clutter/clutter-frame-clock.c b/clutter/clutter/clutter-frame-clock.c
index 4e4238975..6497777e0 100644
--- a/clutter/clutter/clutter-frame-clock.c
+++ b/clutter/clutter/clutter-frame-clock.c
@@ -706,7 +706,7 @@ clutter_frame_clock_notify_ready (ClutterFrameClock *frame_clock)
       break;
     case CLUTTER_FRAME_CLOCK_STATE_DISPATCHED_ONE_AND_SCHEDULED_LATER:
       clutter_frame_clock_set_state (frame_clock,
-                                     CLUTTER_FRAME_CLOCK_STATE_SCHEDULED_NOW);
+                                     CLUTTER_FRAME_CLOCK_STATE_SCHEDULED_LATER);
       maybe_reschedule_update (frame_clock);
       break;
     case CLUTTER_FRAME_CLOCK_STATE_DISPATCHED_TWO:
-- 
2.48.1

